bus-mapping/src/circuit_input_builder.rs

```diff
@@ -188,8 +194,7 @@ pub struct Block {
 
 impl Block {
     /// Create a new block.
-    pub fn new<TX>(
-        _eth_block: &eth_types::Block<TX>,
+    pub fn new(
```

Comment:
I want to share several thoughts:
1. I think we should remove `chain_const` from `Block` and move it to `CircuitInputBuilder`.
2. I think we should not be using `BlockConstants` from `external_tracer`.  The CircuitInputBuilder should not depend on anything from external tracer.  We should add all the fields we need in the `Block` itself.

Reply:
I see, since `external_trace` is only for testing. Let me revert this change.